core/deliverservice/deliveryclient.go

```diff
@@ -128,14 +128,25 @@ func (d *deliverServiceImpl) StartDeliverForChannel(chainID string, ledgerInfo b
 		return errors.Errorf("failed to create block deliverer for channel `%s`, missing OrdererConfig", chainID)
 	}
 
+	createCftDeliverer := false
 	switch ct := oc.ConsensusType(); ct {
 	case "etcdraft":
-		d.blockDeliverer, err = d.createBlockDelivererCFT(chainID, ledgerInfo)
+		createCftDeliverer = true
 	case "BFT":
-		d.blockDeliverer, err = d.createBlockDelivererBFT(chainID, ledgerInfo)
+		switch d.conf.DeliverServiceConfig.Policy {
+		case "cluster", "":
+			d.blockDeliverer, err = d.createBlockDelivererBFT(chainID, ledgerInfo)
+		case "simple":
+			createCftDeliverer = true
+		default:
+			err = errors.Errorf("unexpected delivey service policy: `%s`", d.conf.DeliverServiceConfig.Policy)
+		}
 	default:
 		err = errors.Errorf("unexpected consensus type: `%s`", ct)
 	}
+	if createCftDeliverer {
+		d.blockDeliverer, err = d.createBlockDelivererCFT(chainID, ledgerInfo)
+	}
```

Comment:
why not simply call this function from 2 places? it will save 4 line of code, no? (and would be more readable in my opinion).

Reply:
correct :)